content/pluginsapi/compiler.md

```diff
@@ -3,14 +3,78 @@ title: Compiler
 sort: 2
 ---
 
-## Compiler
+The `Compiler` module of webpack is the main engine that creates a compilation instance with all the options passed through webpack CLI or `webpack` api or webpack comfiguration file.
+
+It is exported by `webpack` api under `webpack.Compiler`.
+
+The compiler is used by webpack by instantiating it and then calling the `run` method. Below is a trivial example of how one might use the `Compiler`. In fact, this is very close to how webpack itself uses it.
+
+[__compiler-example__](https://github.com/pksjce/webpack-internal-examples/blob/master/compiler-example.js)
+
+```javascript
+// Can be imported from webpack package
+import {Compiler} from 'webpack';
+
+// Create a new compiler instance
+const compiler = new Compiler();
+
+// Populate all required options
+compiler.options = {...};
+
+// Creating a plugin.
+class LogPlugin {
+    apply (compiler) {
+        compiler.plugin('should-emit', compilation => {
+            console.log('should i emit');
+            return true;
+        })
+    }
+}
+
+// Apply the compiler to the plugin
+new LogPlugin().apply(compiler);
+
+/* Add other supporting plugins */
+
+// Callback to be executed after run is complete
+const callback = (err, stats) => {
+    console.log('Compiler has finished execution.');
+    /* Do something to show the stats */
+};
+
+// call run on the compiler along with the callback
+compiler.run(callback);
+```
+
+The `Compiler` is what we call a `Tapable` instance. By this, we mean that it mixes in `Tapable` class to imbibe functionality to register and call plugins on itself.
+All user facing plugins are first registered on the `Compiler`.
```

Comment:
Cool! Will do it. You were talking about [this in ResolverFactory](https://github.com/webpack/enhanced-resolve/blob/master/lib/ResolverFactory.js) ?